### DISCUSSION TITLE
How will JEP 451 affect Mockito?

### DISCUSSION THREAD

**Discussion by ascopes** [mockito/mockito#3079] [General]
I was just reading through some of the features that are targeting Java 21, and came across _[JEP 451: Prepare to Disallow the Dynamic Loading of Agents](https://openjdk.org/jeps/451)_.

The proposal discusses plans to "prepare for a future release of the JDK that will, by default, disallow the loading of agents into a running JVM."

This made me wonder about Mocktio, as my very limited (and possibly incorrect) understanding of the framework is that the inline mock maker loads a dynamic ByteBuddy JVM agent that is used to dynamically recompile classes in memory during mocking, enabling it to do stuff like mock final classes, constructors, static attributes, etc.

Curious to know if this is correct and this proposal would potentially affect how Mockito is used in the future, or whether it is totally unrelated to this and I am just misunderstanding. 

If it _did_ affect ByteBuddy and Mockito, that feels like a pretty big downside to this JEP. I'd assume this would potentially affect instrumentation-based systems like AppDynamics as well. If it were a problem, is there a process Mockito would take to raise this with OpenJDK?

**Comment by TimvdLippe** [comment]
There was a previous discussion about this in https://github.com/mockito/mockito/issues/3037 I think the tentative plan is to ask plugins like Surefire to automatically set this flag, as it makes sense that it is only turned on for tests.
